index.js

```diff
@@ -76,8 +77,12 @@ exports.handler = async (event) => {
     }
 
     // return all processed, for easy testing
+    redis.disconnect().catch(err => null)
```

Comment:
So if the lambda is configured with a timeout of 10 seconds... the handler can return earlier than that, but won't be counted as a "success" if a handle stays open.  When it hits 10 seconds of waiting, it counts as an "error" ... and that batch of kinesis records will be re-run.

The `catch` is just extra caution.  Don't really care if it failed to release the redis connection (maybe it never got one, or it exited early?)

Reply:
woops, deleted my earlier comment. 

Hmm, I was referring to the outer try/catch blocks (not the catch method here).  Just wondering how the amazon lambda runtime interacts with the execution of the code in this `try` block (that starts on line 15).

Anyways, I should probably go read the docs! 🤔

Reply:
Ah!  Okay, so it's a bit confusing because I'm doing the async/await thing here, instead of the usual `callback(err)`.

But with a lambda triggered by kinesis ... if you return an error, that "batch" of records is marked as failed, and will be re-run.  If you don't return an error, kinesis moves on to the next "batch".  So generally, in kinesis->lambda, you just have to be careful about what errors you return, to prevent infinitely retrying that batch of records (until they expire in 24 hours).

So I'm trying to explicitly mark retryable errors, and only return/throw/retry those.